src/snowflake/cli/_plugins/spcs/services/commands.py

```diff
@@ -221,20 +225,60 @@ def logs(
     num_lines: int = typer.Option(
         500, "--num-lines", help="Number of lines to retrieve."
     ),
+    since_timestamp: Optional[str] = typer.Option(
+        "", "--since", help="Timestamp to retrieve logs from"
+    ),
+    include_timestamps: bool = typer.Option(
+        False, "--include-timestamps", help="Include timestamps in logs", is_flag=True
+    ),
+    follow: bool = typer.Option(
+        False, "--follow", "-f", help="Continue polling for logs.", is_flag=True
+    ),
+    follow_interval: int = typer.Option(
+        2,
+        "--follow-interval",
+        help="Polling interval in seconds when using the --follow flag",
+    ),
     **options,
 ):
     """
     Retrieves local logs from a service container.
     """
-    results = ServiceManager().logs(
-        service_name=name.identifier,
-        instance_id=instance_id,
-        container_name=container_name,
-        num_lines=num_lines,
-    )
-    cursor = results.fetchone()
-    logs = next(iter(cursor)).split("\n")
-    print_log_lines(sys.stdout, name, "0", logs)
+    if follow:
+        if num_lines != 500:
+            raise IncompatibleParametersError(["--follow", "--num-lines"])
+
+    manager = ServiceManager()
+
+    if follow:
+        stream: Iterable[CommandResult] = (
+            MessageResult(
+                "\n".join(
+                    filter_log_timestamp(log, include_timestamps) for log in log_batch
+                )
+            )
+            for log_batch in manager.stream_logs(
+                service_name=name.identifier,
+                container_name=container_name,
+                instance_id=instance_id,
+                num_lines=num_lines,
+                since_timestamp=since_timestamp,
+                interval_seconds=follow_interval,
+            )
+        )
+        stream = itertools.chain(stream, [MessageResult("")])
+    else:
+        logs = manager.logs(
+            service_name=name.identifier,
+            container_name=container_name,
+            instance_id=instance_id,
+            num_lines=num_lines,
+            since_timestamp=since_timestamp,
+            include_timestamps=include_timestamps,
+        )
+        stream = (MessageResult(extract_log(log)) for log in logs)
```

Comment:
ditto can we move this into `logs` and make the function return a generator of str instead of tuple?